CONTRIBUTING.md

```diff
@@ -129,32 +160,80 @@ Here is a checklist that all Gloo utility crates should fulfill:
       escape hatch for dropping down to raw `web_sys` bindings when an API isn't
       fully supported by the crate yet.
 
-* [ ] There is a loose hierarchy with "mid-level" APIs (which are essentially thin wrappers over the low-level APIs), and "high-level" APIs (which make more substantial changes).
+      Similar for `from_fraw` constructors and `into_raw` conversion methods
+      when applicable.
+
+* [ ] There is a loose hierarchy with "mid-level" APIs (which are essentially
+      thin wrappers over the low-level APIs), and "high-level" APIs (which make
+      more substantial changes).
+
+      As a general rule, the high-level APIs should be built on top of the
+      mid-level APIs, which in turn should be built on top of the low-level APIs
+      (e.g. `web_sys`)
+
+      There are exceptions to this, but they have to be carefully decided on a
+      case-by-case basis.
 
-     As a general rule, the high-level APIs should be built on top of the mid-level APIs, which in turn should be built on top of the low-level APIs (e.g. `web_sys`)
-     
-     There are exceptions to this, but they have to be carefully decided on a case-by-case basis.
+### Implementation and Feedback Cycle
+
+Once we've accepted a design, we can move forward with implementation and
+creating pull requests.
+
+The implementation should generally be unsurprising, since we should have
+already worked out most of the kinks during the earlier design discussions. If
+there are significant new issues or concerns uncovered during implementation,
+then these should be brought up in the design proposal discussion thread again,
+and the evolved design reaffirmed with two team members signing off once
+again.
+
+If there are no new concerns uncovered, then the implementation just needs to be
+checked over by at least one team member. They provide code review and feedback
```

Comment:
I don't think it will be a problem, but I'd like to note that this *does* allow for one team member to approve the pull request while another disapproves.

And this also allows for the team member who approves the pull request to be *different* from the team members who OKed the design.

Reply:
> but I'd like to note that this does allow for one team member to approve the pull request while another disapproves.

I didn't get that impression. You only get to the "Merge" state if all pending feedback has been addressed. So if a team member says "please don't merge this until a test for <foo behaviour>" has been written, that needs to be done before a different team member could approve and merge.

I guess it doesn't handle the (in my mind unlikely) scenario in which a team member is generally concerned about the PR overall. Like if they felt the implementation was just going in completely the wrong direction or something. But even in those cases, I would make the case that nobody should be saying "I *disapprove* of these changes!" without some concrete reason why. And those concrete reasons fall into the Changes Requested state.

Reply:
I was thinking more about the situation with team members being away for a while (or not having enough time to even look at the design).

Like I said though, I don't think it'll be a problem in practice. Just something to keep in mind.

Reply:
I added a note that if any team member has concerns with an implementation, then that must be resolved before merging. I suspect that most of this stuff will already be worked out during the design proposal phase, though.